src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -209,9 +219,39 @@ class BlobCacheServiceImpl implements BlobCacheService {
      * @return The HTTP URI from the cached layer blob is going to be downloaded
      */
     protected String blobDownloadUri(RoutePath route) {
-        StringUtils.pathConcat(blobConfig.baseUrl, route.targetPath)
+        createPresignedGetUrl(blobConfig.storageBucket, route.targetPath)
     }
 
+    /**
+     *  Create a pre-signed URL to download an object in a subsequent GET request.
+     *  @param s3 bucket name
+     *  @param key in the s3 bucket
+     *  @return pre signed URL
+     */
+    private String createPresignedGetUrl(String bucketName, String keyName) {
+        try (S3Presigner presigner = S3Presigner.builder()
+                .region(  Region.of(blobConfig.storageRegion.toUpperCase()) )
+                .credentialsProvider(DefaultCredentialsProvider.create())
+                .build()){
```

Comment:
Wonder if the `presigner` instance can be cached instead of creating a new instance each time

Reply:
Is `S3Presigner` thread-safe? any mention in the docs ?

Reply:
yes, its thread-safe and it is recommended to build it at application startup
https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/presigner/S3Presigner.html